src/ocean/net/util/ParamSet.d

```diff
@@ -151,15 +150,14 @@ class ParamSet
 
     cstring opIndex ( cstring key )
     {
-        try
-        {
-            return this.paramset[this.tolower(key)].val;
-        }
-        catch (ArrayBoundsException e)
-        {
-            e.msg ~= " [\"" ~ key ~ "\"]";
-            throw e;
-        }
+        key = this.tolower(key);
+        auto value = key in this.paramset;
+        verify(
+            value !is null,
+            format("Key '{}' is not found in ParamSet", key)
```

Comment:
Unrelated to this PR, and it's probably not too important, but this allocates, while the `verify` uses reusable instance. Perhaps this could be misused, if the remote party attack us with the requests to hit this path. Is there any way that we format-enable the `Verify` so it can reuse the message buffer?

The answer is probably "no", since we tend not to import anything in `Verify` module.

Reply:
So this is not under control of the app? If yes, I should replace `verify` here with custom exception type. `verify` is expected to never be hit in production, as it basically replaces `assert`.

Reply:
That is a good point. Two applications that I've look into (search for the `QueryParams` and `QueryParamSet`) are fine with this being an assertion, not enforce, since they fill the `ParamSet` with the empty value for every key, then parse the query filling as much values as they can and then they lookup the keys that are guaranteed to be in the set.

Reply:
> So this is not under control of the app?

It is under control of the application. The application knows which keys are in the set. 
The method documentation says, "key must be one of the parameter keys passed on instantiation or added by a subclass." Like AA indexing the application should call this method only if it can for sure pass one of these as a parameter, otherwise it should use `opIn_r`.